### PR TITLE
Bucket list info

### LINKED ISSUE #3616: Output a history JSON file next to buckets when running stellar-core catchup
### Explain in detail the additional functionality you would like to see in stellar-core.

Output a history JSON file next to buckets when running `stellar-core catchup` when run in any mode, such as `--in-memory`. 

### Explain why this feature is important

Stellar core is the only tool that I know of that can construct buckets for any ledger in the history of a network. While it is possible to download buckets from history archives, those buckets will only be for checkpoint ledgers.

The `stellar-core catchup` command is a convenient way to construct a set of buckets that contain a specific ledger's state, but there is no way to conveniently ingest them to render the final state because entries can be duplicated across the buckets. The history JSON is required for understanding what the order of the buckets are.

While some stellar-core commands like `run` store the history JSON in a sqlite db, that is quite an implementation to depend on, and when using `catchup` in isolation the file is never written.

### Describe the solution you'd like

`stellar-core catchup N/M` result in a `buckets/` directory containing the buckets for at least ledger N, which it does today, and a .json file containing history JSON for ledger N.

Any application be able to reconstruct ledger state from the files in `buckets/`.

### Describe alternatives you've considered

Downloading checkpoint buckets from history archives, but doing so would not provide buckets at any ledger.

### Additional context

https://stellarfoundation.slack.com/archives/C0367MGFL/p1669930604478989

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
Resolves #3616 

Adds level and bucket information to the `ledger` section emitted when performing `catchup`.

As the code is shared with the `info` endpoint, I added a new option to `info` as to avoid the verbose output by default.

```
"ledger" : {
         "age" : 526,
         "baseFee" : 100,
         "baseReserve" : 5000000,
         "bucketlist" : [
            [
               "dcba7486da350ee5556e9a41eaef54ae308cc0ebedcc32b35176e430397b3d2d",
               "d0f806f6c944f5bf5d08eb0dd1e44784dd2bb382578b8754361225e9921d1034"
            ],
            [
               "b074145a2978ea067ab9f5f21cb1bceca9c94300b7f8ce86b659ac871b4f99a0",
               "df72dbaab31adf770b8e6bb92aa4740179266ff56d5aa4b997f0c5a2bb75a857"
            ],
            [
               "35a16f226662798080b3ec6fcfdbd87df6b3d73a91626fb9ec6228185317c088",
               "de33acdfb5d22a91e43652efe8236239ccb671fc3e5e9fa4aa2097878ad9a962"
            ],
            [
               "cf96fec89e30eb258075870d0429e4578d5ada1042975211aa16ecc6a2311998",
               "6421c310690dbf68a6cf60243ffae10aa00ccccf6e7a4b8d0d3501f102e83142"
            ],
            [
               "109caf2691cff052d2ba31953e38a559571883fcd07ecceba146042462c62967",
               "3fa689949132b977ff3b298daf8dea6fa379761c661693a2eea4492e489b028b"
            ],
            [
               "c345ad4bb7cce9033d31c8747054b5ab7e7be5da2ffa59f7af700e64ea51da72",
               "c518f1ac76a5948267051cdbedda505b67cdb44be0d17086fddb7b6c2f8abc02"
            ],
            [
               "7a6b15180bc8a20699e38f53294b1c7912a1f9d9c27ae35f974907a74470c8da",
               "984a61244d78a66d22961d68b92c4545d767cdf723a3c8d44c6e4d7e65ffbf3c"
            ],
            [
               "40bcb7d206f648617ebbd499e9ff51761c2f5efe5bf1c244fb21538451dbfc0a",
               "5c932fbc7d7b8fa0681136537a1610230ebeb1eb9b4ecb9629dfe60c1fba9644"
            ],
            [
               "1cad3dcb260d8bceb1fb909fae7d1edc923369056aab0a8b8369b9ca97aa0c21",
               "6756f7a09ebeed43ab5d080774d6d040ac42e074e19f147d1b5b1eae73a35e2f"
            ],
            [
               "d1da9e09b0ce455c06495daf9033a16ae467d6e543c3a3f2a41fd02eb623defd",
               "805c8eacc9ede6cdb9e7882fc484a6608a4a74a160ce5ae05e1f79a5623f421f"
            ],
            [
               "05b98a7ec9dfb14aa2a2aca5495a0e8a6eaf539ea57471b05ffb05780b109403",
               "0000000000000000000000000000000000000000000000000000000000000000"
            ]
         ],
         "closeTime" : 1670032508,
         "hash" : "c8f995d4de473c0d74209797e7bad8eebd8d8fc0c172565bec8f53e9a95c5638",
         "maxTxSetSize" : 105,
         "num" : 1310463,
         "version" : 19
      }
```

### CODE REVIEW

**Comment by leighmcculloch** [comment]
Nice 🎉. That was fast. 

One thing I noticed about the format is that there's some assumed knowledge in the output. Each level has a two length array, which isn't self describing. 

For example:
```json
[
  "dcba7486da350ee5556e9a41eaef54ae308cc0ebedcc32b35176e430397b3d2d",
  "d0f806f6c944f5bf5d08eb0dd1e44784dd2bb382578b8754361225e9921d1034"
]
```

If we gave the two elements names by putting it into an object it'll be self describing:
```json
{
  "curr": "dcba7486da350ee5556e9a41eaef54ae308cc0ebedcc32b35176e430397b3d2d",
  "snap": "d0f806f6c944f5bf5d08eb0dd1e44784dd2bb382578b8754361225e9921d1034"
}
```

(I have no idea if those are curr and snap, I'm just taking a guess at what they should be named.)

**Comment by leighmcculloch** [comment]
Is the way to access this output to pipe it to an application, or to use `--output` to send it to file?

**Comment by MonsieurNicolas** [comment]
re: curr/snap, yeah I agree. I made the change.
To access this, you can simply use the `--output-file` option to `catchup` (this will be written there) or pipe/redirect the output of other commands (like `offline-info`) as stdout contains the json blob (other log lines go to stderr)

**Comment by leighmcculloch** [comment]
This sounds great. 🎉

**Comment by leighmcculloch** [comment]
Is `"num" : 1310463,` in the output the ledger sequence number for the buckets?

**Comment by MonsieurNicolas** [comment]
yes that's it. You check out https://github.com/stellar/stellar-core/blob/master/docs/software/admin.md#general-node-information as this is basically the same thing

**Comment by leighmcculloch** [comment]
Once a bucket shows up in this list, is that bucket file guaranteed to be immutable? Immutable in the sense that the contents won't change on future runs, although I understand that the file could be deleted.

**Comment by MonsieurNicolas** [comment]
yes the name is the hash of the content

File: docs/software/commands.md [hunk 1]
```diff
@@ -184,9 +184,9 @@ format.
   `droppeer?node=NODE_ID[&ban=D]`<br>
   Drops peer identified by NODE_ID, when D is 1 the peer is also banned.
 
-* **info**
+* **info[?compact=false]**
```
[the remark this answers is not in the document]
Comment (leighmcculloch) [reply docs/software/commands.md:187]: +1. The new output seems really small. Is the optional flag useful? It reduces the content by a small number of bytes at a cost of making the ending more complex.

[the remark this answers is not in the document]
Comment (MonsieurNicolas) [reply docs/software/commands.md]: yes this is a typo, it's `compact` by default -> fixed
As for why this flag is needed for the info endpoint: we use it all the time when live monitoring instances from shells so the output needs to fit in a console.

**Comment by marta-lokhova** [comment]
r+ 5fd4df5bddde4b3e6849b6430cf58b0bfe23d476

**Pull request merged**